crates/flux-refineck/src/checker.rs

```diff
@@ -314,26 +307,69 @@ fn check_fn_subtyping(
     // 5. OUTPUT subtyping (f_out <: g_out)
     // RJ: new `at` to avoid borrowing errors...!
     infcx.push_scope();
-    let oblig_output = super_sig
+    let super_output = super_sig
         .output()
         .replace_bound_refts_with(|sort, mode, _| infcx.fresh_infer_var(sort, mode));
     infcx
-        .subtyping(&output.ret, &oblig_output.ret, ConstrReason::Ret)
+        .subtyping(&output.ret, &super_output.ret, ConstrReason::SubtypeOut)
         .with_span(span)?;
+
+    // 6. Update state with Output "ensures" and check super ensures
+    update_ensures(&mut infcx, &mut env, &output, overflow_checking)?;
+    check_ensures(&mut infcx, &mut env, &super_output, ConstrReason::SubtypeEns, span)?;
+
     let evars_sol = infcx.pop_scope().with_span(span)?;
     infcx.replace_evars(&evars_sol);
 
-    if !output.ensures.is_empty() || !oblig_output.ensures.is_empty() {
-        span_bug!(span, "Not yet handled: subtyping with ensures predicates {def_id:?}");
+    Ok(())
+}
+
+fn update_ensures(
+    infcx: &mut InferCtxt,
+    env: &mut TypeEnv,
+    output: &FnOutput,
+    overflow_checking: bool,
+) -> Result {
```

Comment:
This function is infallible
```suggestion
) {
```